src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -101,7 +101,17 @@ public Library GetLibrary(LibraryRange libraryRange, NuGetFramework targetFramew
             // This must exist in the external references
             if (_externalProjectsByUniqueName.TryGetValue(name, out ExternalProjectReference externalReference))
             {
-                packageSpec = externalReference.PackageSpec;
+                // A library with a null version range means that all versions should match.
+                if (externalReference.PackageSpec == null ||
```

Comment:
Do we need this check?

Reply:
@nkolev92 had it in the original PR.  This check guards against null before the `PackageSpec` is dereferenced in the later check `externalReference.PackageSpec.Version`.